sqlest/src/main/scala/sqlest/executor/Database.scala

```diff
@@ -118,29 +118,35 @@ class Session(database: Database) extends Logging {
     withConnection { connection =>
       val (preprocessedSelect, sql, argumentLists) = database.statementBuilder(select)
       val startTime = new DateTime
-      val preparedStatement = prepareStatement(connection, preprocessedSelect, sql, argumentLists)
       try {
-        val resultSet = preparedStatement.executeQuery
+        val preparedStatement = prepareStatement(connection, preprocessedSelect, sql, argumentLists)
         try {
-          val result = extractor(resultSet)
-          val endTime = new DateTime
-          logger.info(s"Ran sql in ${endTime.getMillis - startTime.getMillis}ms: ${logDetails(connection, sql, argumentLists)}")
-          result
+          val resultSet = preparedStatement.executeQuery
+          try {
+            val result = extractor(resultSet)
+            val endTime = new DateTime
+            logger.info(s"Ran sql in ${endTime.getMillis - startTime.getMillis}ms: ${logDetails(connection, sql, argumentLists)}")
+            result
+          } finally {
+            try {
+              if (resultSet != null) resultSet.close
+            } catch {
+              case e: SQLException =>
+            }
+          }
         } finally {
           try {
-            if (resultSet != null) resultSet.close
-          } catch { case e: SQLException => }
+            if (preparedStatement != null) preparedStatement.close
+          } catch {
+            case e: SQLException =>
+          }
         }
       } catch {
         case NonFatal(e) =>
           if (!database.verboseExceptions)
             throw e
           else
             throw new SqlestException(s"Exception running sql: ${logDetails(connection, sql, argumentLists)}", e)
```

Comment:
I was thinking about adding the query and arguments to the SqlestException as named things so you can get them if you want them

Reply:
Yes, that would be a possibility, but it would be a significantly bigger change than fixing what amounts to a bug with the verbose exceptions code.